app/routes/user/password.tsx

```diff
@@ -0,0 +1,250 @@
+              setIsOldPasswordTouched(true)
+            }}
+          />
+          {errorMessage === 'NotAuthorizedException' && (
+            <span className="text-red">Invalid Password</span>
+          )}
+          {errorMessage === 'LimitExceededException' && (
+            <span className="text-red">
+              Attempts Exceeded. Please try again later.
+            </span>
+          )}
+          <Label htmlFor="newPassword">New Password</Label>
+          <InputGroup error={isError}>
+            <TextInput
+              name="newPassword"
+              id="newPassword"
+              className={
+                !isError && isNewPasswordTouched ? 'usa-input--success' : ''
+              }
+              type={visible ? 'text' : 'password'}
+              placeholder="New Password"
+              onChange={(e) => {
+                setNewPassword(e.target.value)
+                setIsNewPasswordTouched(true)
+              }}
+              value={newPassword}
+            />
+            <div className="usa-input-suffix" aria-hidden="true">
+              {visible ? (
+                <Icon.VisibilityOff
+                  className="margin-y-2"
+                  aria-label="Hide password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              ) : (
+                <Icon.Visibility
+                  className="margin-y-2"
+                  aria-label="Show password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              )}
+            </div>
+          </InputGroup>
+          <ButtonGroup>
+            <Button
+              disabled={shouldDisableSubmit}
+              type="submit"
+              className="margin-y-2"
+            >
+              Reset Password
+            </Button>
+          </ButtonGroup>
+          {fetcher.state !== 'idle' && (
+            <>
+              <Spinner /> Saving...
+            </>
+          )}
+          {fetcher.state === 'idle' &&
+            fetcher.data !== undefined &&
+            fetcher.data !== 'NotAuthorizedException' &&
+            fetcher.data !== 'LimitExceededException' && (
+              <>
+                <Icon.Check color="green" /> Saved
+              </>
+            )}
+        </fetcher.Form>
+      </>
+      <div className="usa-alert usa-alert--info usa-alert--validation">
+        <div className="usa-alert__body">
+          <h3 className="site-preview-heading margin-0 usa-alert__heading">
+            New password must contain:
+          </h3>
+          <ul className="usa-checklist" id="validate-code">
+            <li>
+              {containsLower ? (
+                <Icon.Check color="green" aria-label="green check" />
+              ) : (
+                <Icon.Close color="red" aria-label="red x" />
+              )}
+              A lower case letter
+            </li>
```

Comment:
This code is repeated enough times, it's worth factoring into a component. For instance:

```suggestion
            <PasswordValidationStep valid={containsLower}>A lower case letter</PasswordValidationStep>
```